### ISSUE TITLE
"Column index must be at most 0 if positive, not 1" error

### ISSUE THREAD

**Issue by stayingsong** [aryeelab/hichipper#17]
Hello, I tried to run hichipper with a test dataset (JC020 below) and I got the following obscure error at the bottom.  It seems to be related to the qcReport_make.Rmd script, but I can't pinpoint what is happening.  When I run hichipper on the example datasets (e.g. one.yaml), it seems to work.  The input was generated with HiC-Pro, and I deleted all the extraneous files in bowtie_results and hic_results to match the directory structure of the examples.  Would greatly appreciate any insight into why this error is popping up and how one might go about fixing the input.  Thanks!

hichipper --out JC020_out JC020.yaml --skip-diffloop

Thu Jun 01 10:39:00 PDT 2017: Starting hichipper pipeline v0.6.0
Thu Jun 01 10:39:00 PDT 2017: Parsing user parameters
Thu Jun 01 10:54:06 PDT 2017: Performing hichipper-modified background peak calling
Read 6590290 rows and 6 (of 6) columns from 0.271 GB file in 00:00:07
Read 65171550 rows and 4 (of 4) columns from 1.935 GB file in 00:00:23
Read 43535862 rows and 4 (of 4) columns from 1.292 GB file in 00:00:14
Thu Jun 01 11:07:10 PDT 2017: Modified background signal; performing peak calling
Thu Jun 01 11:07:10 PDT 2017: MACS2 TEXT OUTPUT INCOMING
INFO  @ Thu, 01 Jun 2017 11:07:10: Read and build treatment bedGraph... 
INFO  @ Thu, 01 Jun 2017 11:08:23: Read and build control bedGraph... 
INFO  @ Thu, 01 Jun 2017 11:09:08: Build scoreTrackII... 
INFO  @ Thu, 01 Jun 2017 11:09:47: Calculate scores comparing treatment and control by 'qpois'... 
INFO  @ Thu, 01 Jun 2017 11:10:50: Write bedGraph of scores... 
INFO  @ Thu, 01 Jun 2017 11:11:47: Finished 'qpois'! Please check 'JC020_out/hichipper_qvalue.bdg.tmp'! 
INFO  @ Thu, 01 Jun 2017 11:11:47: Read and build bedGraph... 
INFO  @ Thu, 01 Jun 2017 11:12:08: Call peaks from bedGraph... 
INFO  @ Thu, 01 Jun 2017 11:12:09: Write peaks... 
INFO  @ Thu, 01 Jun 2017 11:12:09: Done 
Read 6590290 rows and 6 (of 6) columns from 0.271 GB file in 00:00:08
Anchors removed due to excessive size (likely at ends of chromosomes): 6 
Thu Jun  1 11:12:28 PDT 2017: Processing JC020
head: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/bowtie_results/bwt2/JC020/*pairstat: No such file or directory
Thu Jun  1 11:12:28 PDT 2017: Total_PETs=
cat: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/hic_results/data/JC020/*Pairs: No such file or directory
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_quality_pairs=0
cat: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/hic_results/data/JC020/*_allValidPairs: No such file or directory
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_quality_valid_pairs=0
Thu Jun  1 11:12:28 PDT 2017: Intersecting PETs with anchors
Thu Jun  1 11:12:28 PDT 2017: Finished the anchor merging.
cat: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/hic_results/data/JC020/*Pairs: No such file or directory
cat: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/hic_results/data/JC020/*Pairs: No such file or directory
cat: /Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/hic_results/data/JC020/*_allValidPairs: No such file or directory
Thu Jun  1 11:12:28 PDT 2017: Intrachromosomal_valid_small=0
Thu Jun  1 11:12:28 PDT 2017: Intrachromosomal_valid_med=0
Thu Jun  1 11:12:28 PDT 2017: Intrachromosomal_valid_large=0
Thu Jun  1 11:12:28 PDT 2017: Total number of anchors used: 311
Thu Jun  1 11:12:28 PDT 2017: Total number of reads in anchors: 0
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_intra_quality_anchor=0
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_intra_quality_anchor_small=0
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_intra_quality_anchor_med=0
Thu Jun  1 11:12:28 PDT 2017: Mapped_unique_intra_quality_anchor_large=0
Thu Jun  1 11:12:28 PDT 2017: Loop_PETs=
Thu Jun 01 11:12:28 PDT 2017: 
['Rscript', u'JC020_out/qcReport.R', '/Library/Python/2.7/site-packages/hichipper', 'JC020_out', '/Users/michael/Temporary/hichipper_input', '0.6.0', 'JC020']
/Users/michael/Temporary/hichipper_input

processing file: qcReport_make.Rmd
  |...                                                              |   5%
   inline R code fragments

  |.......                                                          |  11%
label: unnamed-chunk-1 (with options) 
List of 3
 $ echo   : logi FALSE
 $ message: logi TRUE
 $ warning: logi TRUE

  |..........                                                       |  16%
   inline R code fragments

  |..............                                                   |  21%
label: unnamed-chunk-2 (with options) 
List of 3
 $ echo   : logi FALSE
 $ message: logi FALSE
 $ warning: logi FALSE

  |.................                                                |  26%
  ordinary text without R code

  |.....................                                            |  32%
label: unnamed-chunk-3 (with options) 
List of 4
 $ echo   : logi FALSE
 $ message: logi FALSE
 $ warning: logi FALSE
 $ results: chr "asis"

  |........................                                         |  37%
  ordinary text without R code

  |...........................                                      |  42%
label: unnamed-chunk-4 (with options) 
List of 4
 $ echo     : logi FALSE
 $ message  : logi FALSE
 $ warning  : logi FALSE
 $ out.width: chr "\\textwidth"

  |...............................                                  |  47%
  ordinary text without R code

  |..................................                               |  53%
label: unnamed-chunk-5 (with options) 
List of 4
 $ echo     : logi FALSE
 $ message  : logi FALSE
 $ warning  : logi FALSE
 $ out.width: chr "\\textwidth"

  |......................................                           |  58%
  ordinary text without R code

  |.........................................                        |  63%
label: unnamed-chunk-6 (with options) 
List of 6
 $ echo     : logi FALSE
 $ message  : logi FALSE
 $ warning  : logi FALSE
 $ results  : chr "asis"
 $ out.width: chr "\\textwidth"
 $ fig.width: num 7

Quitting from lines 166-189 (qcReport_make.Rmd) 
Error in { : 
  task 1 failed - "Column index must be at most 0 if positive, not 1"
Calls: <Anonymous> ... withCallingHandlers -> withVisible -> eval -> eval -> %do% -> <Anonymous>

**Comment by stayingsong** [comment]
Nevermind, I realized the obvious path bugs, will try to fix those first!

**Comment by caleblareau** [comment]
@stayingsong Could you describe the issue a bit more? I'd like to catch the issue before it runs through the whole pipeline. Did you point to a directory that doesn't exist or what exactly?

**Comment by stayingsong** [comment]
@caleblareau Apologies for the late response, so what I think happened was that I specified absolute rather than relative paths in the YAML configuration file for hichipper.  When I used absolute paths to specify the locations of the restriction digest file, the input file, etc. in the YAML file, the pipeline would actually append the path to the path of the directory in which I ran the hichipper command.

For example, if I was running hichipper in this directory: /Users/michael/Temporary/hichipper_input/

And specified that my input files from Hi-C-Pro were in the following folder: /Users/Michael/Temporary/hichipper_input/JC020/bowtie_results/bwt2/JC020/

Then it would fail to find the required input files, for example:

/Users/michael/Temporary/hichipper_input//Users/Michael/Temporary/hichipper_input/JC020/bowtie_results/bwt2/JC020/*pairstat: No such file or directory

And the resulting tables in R would have 0 rows/columns, resulting in the various downstream errors I initially observed.  I had actually discovered the finickiness with the paths before, but I must have accidentally reverted them to absolute paths during testing and forgot.
